--- origsrc/dillo-0.8.6/dpi/Makefile.am	2006-01-03 14:16:10.000000000 -0600
+++ src/dillo-0.8.6/dpi/Makefile.am	2006-06-28 19:31:59.375000000 -0500
@@ -18,18 +18,18 @@
 cookies_PROGRAMS = cookies.dpi
 datauri_PROGRAMS = datauri.filter.dpi
 
-bookmarks_dpi_LDADD = @GLIB_LIBS@ ../dpip/libDpip.a
+bookmarks_dpi_LDADD = ../dpip/libDpip.a @GLIB_LIBS@
 if DLGUI
-downloads_dpi_LDADD = @GLIB_LIBS@ @LIBFLTK_LIBS@ ../dpip/libDpip.a
+downloads_dpi_LDADD = ../dpip/libDpip.a @GLIB_LIBS@ @LIBFLTK_LIBS@
 else
-downloads_dpi_LDADD = @GLIB_LIBS@ ../dpip/libDpip.a
+downloads_dpi_LDADD = ../dpip/libDpip.a @GLIB_LIBS@
 endif
-ftp_filter_dpi_LDADD = @GLIB_LIBS@ ../dpip/libDpip.a
-https_filter_dpi_LDADD = @GLIB_LIBS@ @LIBSSL_LIBS@ ../dpip/libDpip.a
-hello_filter_dpi_LDADD = @GLIB_LIBS@ ../dpip/libDpip.a
-file_dpi_LDADD = @GLIB_LIBS@ @LIBPTHREAD_LIBS@ ../dpip/libDpip.a
-cookies_dpi_LDADD = @GLIB_LIBS@ ../dpip/libDpip.a
-datauri_filter_dpi_LDADD = @GLIB_LIBS@ ../dpip/libDpip.a
+ftp_filter_dpi_LDADD = ../dpip/libDpip.a @GLIB_LIBS@
+https_filter_dpi_LDADD = ../dpip/libDpip.a @GLIB_LIBS@ @LIBSSL_LIBS@
+hello_filter_dpi_LDADD = ../dpip/libDpip.a @GLIB_LIBS@
+file_dpi_LDADD = ../dpip/libDpip.a @GLIB_LIBS@ @LIBPTHREAD_LIBS@
+cookies_dpi_LDADD = ../dpip/libDpip.a @GLIB_LIBS@
+datauri_filter_dpi_LDADD = ../dpip/libDpip.a @GLIB_LIBS@
 
 file_dpi_LDFLAGS = @LIBPTHREAD_LDFLAGS@
 
--- origsrc/dillo-0.8.6/dpid/Makefile.am	2005-05-07 14:28:00.000000000 -0500
+++ src/dillo-0.8.6/dpid/Makefile.am	2006-06-28 19:29:01.906250000 -0500
@@ -2,7 +2,7 @@
 AM_CFLAGS = @GLIB_CFLAGS@
 
 bin_PROGRAMS = dpid
-dpid_LDADD = @GLIB_LIBS@ ../dpip/libDpip.a
+dpid_LDADD = ../dpip/libDpip.a $(GLIB_LIBS)
 
 EXTRA_DIST = dpidc
 bin_SCRIPTS = dpidc
